morse.rb

```diff
@@ -1,15 +1,60 @@
-def mouse(mourse_code)
-  codes = { '.-' => 'A', '-...' => 'B', '-.-.' => 'C',
-            '-..' => 'D', '.' => 'E', '..-.' => 'F', '--.' => 'G',
-            '....' => 'H', '..' => 'I', '.---' => 'J', '-.-' => 'K',
-            '.-..' => 'L', '--' => 'M', '-.' => 'N', '---' => 'O',
-            '.--.' => 'P', '--.-' => 'Q', '.-.' => 'R', '...' => 'S',
-            '-' => 'T', '..-' => 'U', '...-' => 'V', '.--' => 'W',
-            '--..' => 'Z', '-..-' => 'X', '-.--' => 'Y' }
-  mourse_code
-    .split('   ')
-    .map do |item|
-      item.split.map { |char| codes[char] }.join
-    end
-    .join(' ')
+def decode_char(code)
+  characters = {
+    '.-': 'A',
+    '-...': 'B',
+    '-.-.': 'C',
+    '-..': 'D',
+    '.': 'E',
+    '..-.': 'F',
+    '--.': 'G',
+    '....': 'H',
+    '..': 'I',
+    '.---': 'J',
+    '-.-': 'K',
+    '.-..': 'L',
+    '--': 'M',
+    '-.': 'N',
+    '---': 'O',
+    '.--.': 'P',
+    '--.-': 'Q',
+    '.-.': 'R',
+    '...': 'S',
+    '-': 'T',
+    '..-': 'U',
+    '...-': 'V',
+    '.--': 'W',
+    '-..-': 'X',
+    '-.--': 'Y',
+    '--..': 'Z',
+    '.----': '1',
+    '..---': '2',
+    '...--': '3',
+    '....-': '4',
+    '.....': '5',
+    '-....': '6',
+    '--...': '7',
+    '---..': '8',
+    '----.': '9',
+    '-----': '0'
+  }
+
+  characters[code.to_sym]
 end
+
+def decode_word(ciphered)
+  word = ''
+  ciphered.split.each do |code|
+    word += decode_char(code)
+  end
+  word
+end
+
+def decrypt(ciphered)
+  sentence = ''
+  ciphered.split('  ').each do |word|
```

Comment:
your team does not have a method that decodes single characters of the morse code 👍🏾  please work on this and submit again